Explorer/Controller/CreateLogViewController.swift

```diff
@@ -152,9 +163,53 @@ class CreateLogViewController: UIViewController {
     @objc func dismissInputView() {
         self.view.endEditing(true)
         
+        dismissPickerView()
+    }
+    
+    deinit {
+        NotificationCenter.default.removeObserver(self)
+    }
+
+    @objc func keyboardWillShow(notification: Notification) {
+        dismissPickerView()
+        
+        guard let userInfo = notification.userInfo else { return }
+        guard let keyboardSize = userInfo[UIResponder.keyboardFrameEndUserInfoKey] as? NSValue else { return }
+        let keyboardFrame = keyboardSize.cgRectValue
+        inputOffset = view.frame.height - (keyboardFrame.height + 18)
+        
+        var keyboardFocus: CGFloat = 0.0
+        
+        if createLogView.diveNumberTextField.isFirstResponder {
+            keyboardFocus = createLogView.diveNumberTextField.frame.origin.y
+        }
+
+        if createLogView.locationTextField.isFirstResponder {
```

Comment:
Consider using `else if` for these subsequent checks. It's not really a bit deal, but if `createLogView.diveNumberTextField.isFirstResponder == true` then you really don't need to check these other `if/else` cases.

Reply:
Done!